Improve the stdlib for the calculator + extra special commands

sayaks opened the pull request:
This means adding `in`, `flatten`, `remove`, and `interleave` for lists, as well as adding assoc-lists and sets.

Assoc lists are lists of pairs, they act like a dictionary but implemented as a list of pairs.
Sets are lists with only unique elements where the order doesn't matter.

I also added a timing function and timeout toggle for running the calculator in the terminal.

DXsmiley commented:
These look pretty good. However if you could add some tests for the new calclib functions that'd be great. Just add some things to the bottom of `tests/test_calculator.py`.

sayaks commented:
Right, now i've added tests and also fixed a mistake in the README i noticed. (you said you need to do `python bot.py parameters.json` when in reality you have to do `python entrypoint.py parameters.json`)
